Tallies/TallyResponses/densityResponse_class.f90

```diff
@@ -0,0 +1,96 @@
+module densityResponse_class
+
+  use numPrecision
+  use universalVariables,  only : neutronMass, lightSpeed
+  use dictionary_class,    only : dictionary
+  use particle_class,      only : particle, P_NEUTRON, P_PHOTON
+  use tallyResponse_inter, only : tallyResponse
+
+  ! Nuclear Data interface
+  use nuclearDatabase_inter, only : nuclearDatabase
+
+  implicit none
+  private
+
+  !!
+  !! tallyResponse to score particle density contribution
+  !!
+  !! Returns the inverse of the particle velocity in [cm/s]
+  !!
+  !! The velocity is calculated from the particle energy for neutrons, and it is
+  !! the speed of light for photons
+  !!
+  !! Interface:
+  !!   tallyResponse Interface
+  !!
+  type, public,extends(tallyResponse) :: densityResponse
+    private
+  contains
+    procedure :: init
+    procedure :: get
+    procedure :: kill
+  end type densityResponse
+
+contains
+
+  !!
+  !! Initialise Response from dictionary
+  !!
+  !! See tallyResponse_inter for details
+  !!
+  subroutine init(self, dict)
+    class(densityResponse), intent(inout) :: self
+    class(dictionary), intent(in)         :: dict
+
+    ! Do nothing
+
+  end subroutine init
+
+  !!
+  !! Calculates the particle velocity for neutron and photons
+  !! NOTE: neutronMass: [MeV]
+  !!       lightSpeed:  [cm/s]
+  !!
+  !! The function returns the inverse of the velocity (response to score particle density)
+  !! if the particle type is neutron or photon, and zero otherwise
+  !!
```

Comment:
One thing I might suggest is moving the speed calculation to the particle since we want it anyway (e.g., for alpha, dynamic), but this may not be necessary for the moment.

Reply:
Good point! A method in the particle may be the best place for the velocity indeed! 
But I second that it is not necessary at the moment. We can marge this PR as it is and open an Issue about it. It would be a good 'first issue' for new contributors do deal with ;-)

Reply:
For me, I don't think that's necessary right now. It might clutter the particle and it won't be used for a long time. But it's a small change so I'll respect democracy.

Reply:
In the end I went for it, it was a very fast change. I believe I also addressed all the comments, so let me know if I can go ahead and merge.